File: ContainsDuplicate.java

import java.util.HashSet;




class ContainsDuplicate {
    public static Boolean containsDuplicate(int[] nums) {
        HashSet<Integer> seen = new HashSet<>();
        
        for(int num: nums) {
            if(seen.contains(num)) {
                return true;
            } else {
                seen.add(num);
            }
        }
        return false;
        
    }
    public static void main(String[] args) {
        int[] nums = {1, 2, 3, 1};
        System.out.println(containsDuplicate(nums));
    }
}

File: ContainsDuplicate.py




def contains_duplicate(nums: List[int]) -> bool:
    if(len(nums) != len(set(nums))): 
        print("False")
        return False
    else: 
        print("True")
        return True

if __name__ == "__main__":
    raise SystemExit(contains_duplicate([1,2,3,1]))
